samples/javascript_nodejs/23.facebook-events/bots/facebookBot.js

```diff
@@ -0,0 +1,199 @@
+                if (context.activity.channelId !== "facebook") {
+                    await context.sendActivity("This sample is intended to be used with a Facebook bot.");
+                }
+                await this.showChoices(context);
+            }
+        });
+
+
+
+        this.onEvent(async (context) => {
+            this.logger.log('Processing an Event Activity.');
+
+            // Analyze Facebook payload from EventActivity.Value
+            await this.processFacebookPayload(context, context.activity.value);
+
+        });
+
+
+    }
+
+    /**
+     * 
+     * @param {TurnContext} context 
+     */
+    async showChoices(context) {
+        // Create choices for the prompt
+        const choices = [
+            {
+                value: quickRepliesOption
+            },
+            {
+                value: facebookPageIdOption
+            },
+            {
+                value: postBackOption
+            }
+        ]
+        
+        // Create the prompt message
+        var message = ChoiceFactory.forChannel(context.activity.channelId, choices, "What Facebook feature would you like to try? Here are some quick replies to choose from!");
+        await context.sendActivity(message);
+    };
+
+    /**
+     * @param {TurnContext} context 
+     * @param {Object} data
+     */
+
+    async processFacebookPayload(context, data) {
+
+        // At this point we know we are on Facebook channel, and can consume the Facebook custom payload present in channelData.
+        const facebookPayload = data;
+        if (facebookPayload) {
+
+            //Postback
+            if (facebookPayload.postback) {
+                await this.onFacebookPostback(context, facebookPayload.postback);
+                return true;
+            }
+
+            //Optin
+            else if (facebookPayload.optin) {
+                await this.onFacebookOptin(context, facebookPayload.optin);
+                return true;
+            }
+
+            //Quick Reply
+            else if (facebookPayload.message && facebookPayload.message.quick_reply) {
+                await this.onFacebookQuickReply(context, facebookPayload.message.quick_reply);
+                return true;
+            }
+
+
+            //Echo
+            else if (facebookPayload.message && facebookPayload.message.is_echo) {
+                await this.onFacebookEcho(context, facebookPayload.message);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param {Object} postback JSON object for postback payload.
+     * @param {TurnContext} context
+     */
+
+    async onFacebookPostback(context, postback) {
+        this.logger.log('Postback message received.');
+        // TODO: Your postBack handling logic here...
+
+        //Answer the postback and show choices
+        await context.sendActivity("Are you sure?");
+        await this.showChoices(context);
+    }
+
+    /**
+     * @param {Object} optin 
```

Comment:
remove xml comments from method

Reply:
Added this:
/**
     * Called when receiving a Facebook messaging_optin event.
     * Facebook Developer Reference: Optin
     * https://developers.facebook.com/docs/messenger-platform/reference/webhook-events/messaging_optins/
     * @param {Object} optin